src/CiviEntityStorage.php

```diff
@@ -619,4 +620,60 @@ protected function saveToDedicatedTables(ContentEntityInterface $entity, $update
     }
   }
 
+  /**
+   * Allows CiviCRM hook to invoke presave.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to save.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   *   If the entity identifier is invalid.
+   *
+   * @see \Drupal\Core\Entity\ContentEntityStorageBase::doPreSave
+   */
+  public function civiPreSave(EntityInterface $entity) {
+    $this->doPreSave($entity);
+  }
+
+  /**
+   * Allows CiviCRM hook to invoke postsave.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The saved entity.
+   * @param $update
+   *   Specifies whether the entity is being updated or created.
+   *
+   * @see \Drupal\Core\Entity\ContentEntityStorageBase::doPostSave
+   */
+  public function civiPostSave(EntityInterface $entity, $update) {
+    $this->doPostSave($entity, $update);
+  }
+
+  /**
+   * Allows CiviCRM hook to invoke predelete.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to be deleted.
+   *
+   * @see \Drupal\Core\Entity\EntityStorageInterface::delete
+   */
+  public function civiPreDelete(EntityInterface $entity) {
+    CivicrmEntity::preDelete($this, [$entity]);
+    $this->invokeHook('predelete', $entity);
+  }
```

Comment:
The base entity storage classes don't have a clean pre/post delete handling (like the pre/post save) so these `::civiPreDelete` and `::civiPostDelete` have to do a bit more work than just passing off to existing code.